ln1_f4.py

```diff
@@ -1,115 +1,108 @@
 mem_total = int(raw[7])
 mem_used = int(raw[8])
 mem_free = int(raw[9])
 mem_cached = int(raw[12])
-mem_string = ("MEMORY - Total:%sMb, Used:%sMb, Free:%sMb, Cached:%sMb" % (mem_total, mem_used, mem_free, mem_cached))
-print (mem_string)
+print ("MEMORY - Total:%sMb, Used:%sMb, Free:%sMb, Cached:%sMb" % (mem_total, mem_used, mem_free, mem_cached))
 
-raw = get_raw_data("df -m --total") #обробка данних по жорсткому диску
+raw = get_sysresinfo("df -m --total")  # обробка данних по жорсткому диску
 hdd_total = int(raw[50])
 hdd_used = int(raw[51])
 hdd_free = int(raw[52])
-hdd_string = ("Hard disk drive - Total:%sMb, Used:%sMb, Free:%sMb" % (hdd_total, hdd_used, hdd_free))
-print (hdd_string)
+print ("Hard disk drive - Total:%sMb, Used:%sMb, Free:%sMb" % (hdd_total, hdd_used, hdd_free))
 
-                        #формування єдиного рядка з данними для запису в файл
-var_names_list = [cpu_total,cpu_user,cpu_sys,cpu_idle,mem_total,mem_used,mem_free,mem_cached,hdd_total,hdd_used,hdd_free]
-for x in var_names_list:
-    data_string.append(x)
+systime_customformat = datetime.datetime.now()
+systime_customformat = systime_customformat.strftime("%d,%m,%Y,%H,%M,%S")
 
-###############################################################################################################
-
-try:
-    database_file = open("database.txt", "r") #пробую відкрити файл БД
-    db_lines_count = int(database_file.readline())
+sysinfo_variables_names = [cpu_total, cpu_user, cpu_sys, cpu_idle, mem_total, mem_used, mem_free,
+                           mem_cached, hdd_total, hdd_used, hdd_free]
+sysinfo_database_keys = ["cpu_total", "cpu_user", "cpu_sys", "cpu_idle", "mem_total", "mem_used", "mem_free",
+                         "mem_cached", "hdd_total", "hdd_used", "hdd_free"]
 
-except IOError:
-    database_file = open("database.txt", "w") #якщо помилка то створюю чистий
-    database_file.write("0"+"\n")
-    db_lines_count = 0
-
-finally:
-    database_file.close() #закриваю файл
+###############################################################################################################
 
-database_file = open("database.txt", "a") #відкриваю файл та добавляю рядок з данними в БД
-database_file.write(str(data_string) + "\n")
-database_file.close()
-db_lines_count += 1
-database_file = open("database.txt", "r+") #перезаписую лічильник кількості рядків в БД
-database_file.seek(0,0)
-database_file.write(str(db_lines_count) + "\n")
-database_file.close()
+if os.path.isfile("database2") == False:
+    with open("database2", "w") as database_file:
+        sysinfo_database = {}
+        database_file.write(json.dumps(sysinfo_database))
+
+with open("database2", "r") as database_file:
+    data = database_file.read().strip()
+sysinfo_database = json.loads(data)
+sysdb_totalrecords = len(sysinfo_database) + 1
+sysinfo_database[str(sysdb_totalrecords)] = {
+    "record_time": systime_customformat,  # saving time but not using at the moment (plan in future)
```

Comment:
Yes but it'll make algorithm far more complicated). If timestamp used, you will have problems with time zone changes, winter/summer time or just corecting the system clock, which will break records chronology. With strict order number as key you don't have such problem (I had this problem with one of projects in past). Still I agree with you and i'll try to redo argorithm/